### PR TITLE
Use latest directory commit timestamp as tag

### PR DESCRIPTION
**Pull request by Artyop**
This commit change the way of tagging image, from using git-lfs-tree sha256 to using directory's last commit timestamp. It will permit renovate to sort versions and update properly new images, where renovate wasn't able to do so before due to its inability to sort SHAs.
The git log command, when given a specific directory in parameter, retrieve the last commit were changes were made in said directory. This means that we can ensure different image versions same as used to be with git-tree.
Renovate can sort versions from left to right, using a timestamp as a tag value allows it to retrieve the latest version deployed with ease.

### CODE REVIEW

**Comment by Artyop** [comment]
Bit of an issue with libc version while creating the images it seems  
`ImportError: /lib64/libc.so.6: version `GLIBC_2.34' not found (required by /lib64/libgcc_s.so.1)` ` when trying to use dnf for anything other than an install, looking into it

**Comment by sayboras** [comment]
Can you rebase with main branch to pick up https://github.com/cilium/image-tools/pull/281 to avoid any potential surprise?

**Comment by Artyop** [comment]
Rebased from main branch, checking for the key issue on the iptables-20.04-image

**Comment by sayboras** [comment]
> checking for the key issue on the iptables-20.04-image

iptables-20.04-image might not be used anymore, as we moved to snapshot build for specific iptables version.

https://github.com/cilium/image-tools/pull/287

**Comment by sayboras** [comment]
#287 is merged, so you might need to rebase again 😅 , good news is no need to check for key issue 🔑

**Comment by Artyop** [comment]
> #287 is merged, so you might need to rebase again 😅 , good news is no need to check for key issue 🔑

well that sounds like a good deal to me!

**Review by michi-covalent — APPROVED** [review]
it might still be useful to have commit sha as a part of the tag (like $timestamp-$sha or something) so that it's easy to see which commit the image got built from.

**Comment by Artyop** [comment]
@michi-covalent added the short commit right after the timestamp separated with a "-", let me know if it sounds good for you

**Review by michi-covalent — APPROVED** [review]
🥰

**Review by nbusseneau — APPROVED** [review]

**Comment by joestringer** [comment]
Hi, could this have broken the way that we ensure tags are unique based on the content of the image?

This month, after merging https://github.com/cilium/image-tools/commit/b6afb9dd5e0102e41d4e936ecd65a49431eb1109 and https://github.com/cilium/image-tools/commit/3e3093f70de87f4881c166dc3ef44f2ba11fdb24, the existing tag `1755531540-60ee83e` was overwritten instead of generating a new tag. This seems to be the first time that changes in the repository did not generate a new tag. One of the surprising aspects is that the suffix is 60ee83e and not b6afb9d, despite the new version of the script intending to include a short commit hash in the tag.

image

**Comment by joestringer** [comment]
I tracked back to this PR by checking when we changed the scheme for the docker tags, however now I think a bit more it's clear that this couldn't have introduced such a breakage as this change was made over a year ago and we only encountered problems in the last few weeks. I'll search for a newer PR that may have broken the behavior more recently.

**Comment by HadrienPatte** [comment]
👋 The issue we're experiencing is described in more details in #385 (and there's a pending fix for cilium/cilium in https://github.com/cilium/cilium/pull/41710

**Pull request merged**
